appcues/src/main/java/com/appcues/trait/appcues/ModalTrait.kt

```diff
@@ -23,17 +28,21 @@ import org.koin.core.scope.Scope
 
 internal class ModalTrait(
     override val config: AppcuesConfigMap,
-    private val renderContext: RenderContext,
-    private val scope: Scope,
+    renderContext: RenderContext,
+    scope: Scope,
 ) : ContentWrappingTrait, PresentingTrait {
 
     companion object {
 
         const val TYPE = "@appcues/modal"
     }
 
-    // should we have a default presentation style?
-    private val presentationStyle = config.getConfigOrDefault("presentationStyle", "full")
+    private enum class PresentationStyle {
+        DIALOG, FULL, HALF_SHEET, SHEET
+    }
+
+    // this can throw on init, if invalid presentation style - not one of the known types
+    private val presentationStyle = config.getConfig<String?>("presentationStyle").toPresentationStyle()
```

Comment:
this is the key check now, preventing the error 1 from the description -- it will throw right away (L94) and be caught and handled with error event